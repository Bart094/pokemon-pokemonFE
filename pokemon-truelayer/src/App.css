.App-logo {
  height: 100px;
}

.App-header {
  background-color: #808080;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  text-align: center;
}

.App-input {
  background-color: rgb(247, 171, 27);
  border-radius: 1em;
  height: 20px;
  text-align: center;
  color: white;
  font-size: 14px;
}

.App-input:focus {
  background-color: red;
}

.App-button {
  height: 50px;
  vertical-align: middle;
}

*:focus {
  outline: none;
}

