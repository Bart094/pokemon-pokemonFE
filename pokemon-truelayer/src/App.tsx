import "./App.css";
import React from "react";
import Pokemon from "./pokemon";
import Favourites from "./favourites";

export default function App() {
  const [showPokemon, setShowPokemon] = React.useState<boolean>(false);
  const [showFavourites, setShowFavourites] = React.useState(false);
  const [pokemonName, setPokemonName] = React.useState("");
  const [disableButton, setDisableButton] = React.useState(true);

  function handleSubmit(event: any) {
    event.preventDefault();
    setPokemonName(event.target[0].value);
    setShowPokemon(true);
  }

  function handleChange(input: any) {
    setPokemonName(input.currentTarget.value);
    if (pokemonName.length > 0) {
      setDisableButton(false);
    } else {
      setDisableButton(true);
    }
  }

  return (
    <div className="App-header">
      {showPokemon == false && showFavourites == false && (
        <div>
          <img
            className="App-logo"
            src="https://truelayer.com/static/f7ff55b4588b66f31ccc555a48bdf186/truelayer_logo_white_w167_h31.svg"
          ></img>
          <form action="#" onSubmit={handleSubmit}>
            <input
              className="App-input"
              type="text"
              name="name"
              placeholder="Insert Pokemon Name"
              onChange={handleChange}
            />
            <input
              disabled={disableButton}
              className="App-button"
              type="image"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/53/Pok%C3%A9_Ball_icon.svg/512px-Pok%C3%A9_Ball_icon.svg.png"
            />
          </form>
          <button
            className="App-input"
            type="button"
            onClick={() => setShowFavourites(true)}
          >
            List Of Favourites
          </button>
        </div>
      )}
      {showFavourites ? <Favourites /> : null}
      {showPokemon ? <Pokemon pokemonName={pokemonName.toLowerCase()} /> : null}
      {showPokemon || showFavourites ? (
        <button
          className="App-input"
          type="submit"
          autoFocus={true}
          onKeyPress={() => setShowPokemon(!showPokemon)}
          onClick={() => {
            if (showFavourites || showPokemon) {
              setShowPokemon(false);
              setShowFavourites(false);
            }
            setDisableButton(true);
          }}
        >
          Back
        </button>
      ) : null}
    </div>
  );
}
