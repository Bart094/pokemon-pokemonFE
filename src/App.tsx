import "./App.css";
import React, { BaseSyntheticEvent, FormEvent } from "react";
import Pokemon from "./components/pokemon";
import Favourites from "./components/favourites";

export default function App() {
  const [showPokemon, setShowPokemon] = React.useState<boolean>(false);
  const [showFavourites, setShowFavourites] = React.useState(false);
  const [pokemonName, setPokemonName] = React.useState("");
  const [disableButton, setDisableButton] = React.useState(true);

  function handleSubmit(event: any) {
    event.preventDefault();
    setPokemonName(event.currentTarget[0].value);
    setShowPokemon(true);
  }

  function handleChange(input: any) {
    if (input.currentTarget.value.length > 0) {
      setDisableButton(false);
    } else {
      setDisableButton(true);
    }
  }

  return (
    <div className="App-header">
      {showPokemon === false && showFavourites === false && (
        <div>
          <img
            className="App-logo"
            src={require('./images/truelayer_logo.svg').default}
            alt="logo-truelayer"
            data-testid="logo"
          ></img>
          <form action="#" onSubmit={handleSubmit}>
            <input
              className="App-input"
              type="text"
              name="name"
              placeholder="Insert Pokemon Name"
              onChange={handleChange}
              data-testid="input-text"
            />
            <input
              disabled={disableButton}
              className="App-button"
              type="image"
              src={require('./images/ball_icon.png')}
              alt="poke-ball"
              data-testid="input-button"
            />
          </form>
          <button
            className="App-input"
            type="button"
            onClick={() => setShowFavourites(true)}
          >
            List Of Favourites
          </button>
        </div>
      )}
      {showFavourites ? <Favourites /> : null}
      {showPokemon ? <Pokemon pokemonName={pokemonName.toLowerCase()} /> : null}
      {showPokemon || showFavourites ? (
        <button
          className="App-input"
          type="submit"
          autoFocus={true}
          onClick={() => {
            if (showFavourites || showPokemon) {
              setShowPokemon(false);
              setShowFavourites(false);
            }
            setDisableButton(true);
          }}
        >
          Back
        </button>
      ) : null}
    </div>
  );
}
