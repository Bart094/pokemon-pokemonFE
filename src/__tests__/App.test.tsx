import App from "../App";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";


Enzyme.configure({ adapter: new Adapter() });

const mockSubmitEvent: any = {
  currentTarget: {
    0: {
      value: "articuno",
    },
  },
  preventDefault: () => {},
};

const mockChangeEvent: any = {
  currentTarget: {
    value: "articuno",
  },
};

const mockClearEvent: any = {
  currentTarget: {
    value: "",
  },
};


describe("search Pokemon", () => {
  it("should pass shallow snapshot test", () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find(".App-input").length).toBe(2);
    wrapper.find(".App-input").at(0).simulate("change", mockChangeEvent);
    wrapper.find(".App-input").at(0).simulate("change", mockClearEvent);
    wrapper.find("form").simulate("submit", mockSubmitEvent);
  });
});

describe("renders <Favourites/>", () => {
  it("should pass shallow snapshot test", () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find("button").simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find("Favourites").length).toBe(1);
    expect(wrapper.find("button").length).toBe(1);
    wrapper.find("button").simulate("click");
  });
});
