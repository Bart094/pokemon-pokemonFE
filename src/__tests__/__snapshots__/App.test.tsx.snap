// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Favourites/> should pass shallow snapshot test 1`] = `
<div
  className="App-header"
>
  <div>
    <img
      alt="logo-truelayer"
      className="App-logo"
      data-testid="logo"
      src="truelayer_logo.svg"
    />
    <form
      action="#"
      onSubmit={[Function]}
    >
      <input
        className="App-input"
        data-testid="input-text"
        name="name"
        onChange={[Function]}
        placeholder="Insert Pokemon Name"
        type="text"
      />
      <input
        alt="poke-ball"
        className="App-button"
        data-testid="input-button"
        disabled={true}
        src="ball_icon.png"
        type="image"
      />
    </form>
    <button
      className="App-input"
      onClick={[Function]}
      type="button"
    >
      List Of Favourites
    </button>
  </div>
</div>
`;

exports[`renders <Favourites/> should pass shallow snapshot test 2`] = `
<div
  className="App-header"
>
  <Favourites />
  <button
    autoFocus={true}
    className="App-input"
    onClick={[Function]}
    type="submit"
  >
    Back
  </button>
</div>
`;

exports[`search Pokemon should pass shallow snapshot test 1`] = `
<div
  className="App-header"
>
  <div>
    <img
      alt="logo-truelayer"
      className="App-logo"
      data-testid="logo"
      src="truelayer_logo.svg"
    />
    <form
      action="#"
      onSubmit={[Function]}
    >
      <input
        className="App-input"
        data-testid="input-text"
        name="name"
        onChange={[Function]}
        placeholder="Insert Pokemon Name"
        type="text"
      />
      <input
        alt="poke-ball"
        className="App-button"
        data-testid="input-button"
        disabled={true}
        src="ball_icon.png"
        type="image"
      />
    </form>
    <button
      className="App-input"
      onClick={[Function]}
      type="button"
    >
      List Of Favourites
    </button>
  </div>
</div>
`;
